docs/plugin/repository.md

```diff
@@ -0,0 +1,219 @@
+   go mod tidy
+   ```
+
+3. Build all your plugins. See [**Build plugin binaries**](custom-executor.md).
+
+4. Generate an index file:
+
+   ```bash
+   go run github.com/kubeshop/botkube/hack -binaries-path "./dist" -url-base-path "https://example.com"
+   ```
+
+   :::info
+   Replace the `-url-base-path` flag with the base path of your HTTP server. See [**Hosting plugins**](#host-plugins) for some examples.
+   :::
+
+# Host plugins
+
+This section describes example ways for serving Botkube plugins.
+
+## GitHub releases
+
+A GitHub release allows you to upload additional assets that are later accessible with a predictable URL. When you generate the index file, specify the `-url-base-path` flag as `https://github.com/{owner}/{repo}/releases/download/{release_tag}`, for example, `https://github.com/kubeshop/botkube/releases/download/v1.0.0`.
+
+Once the plugin binaries are built and the index file is generated, you can create a GitHub release using [GitHub CLI](https://cli.github.com/). For example:
+
+```bash
+gh release create v1.0.0 \
+ ./dist/source_* \
+ ./dist/executor_* \
+ ./plugins-index.yaml
+```
+
+You can also use [GitHub Actions](https://docs.github.com/en/actions) to publish Botkube plugins automatically each time a new tag is pushed. To do that, add the following workflow under `.github/workflows/release.yml` in your repository:
+
+```yaml
+name: Official Release
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: "go.mod"
+          cache: true
+
+      - name: Build plugins binaries
+        uses: goreleaser/goreleaser-action@v3
+        with:
+          version: latest
+          args: build --rm-dist
+
+      - name: Generate plugins index.yaml
+        env:
+          PLUGIN_DOWNLOAD_URL_BASE_PATH: "https://github.com/${{ github.repository }}/releases/download/${{ github.ref_name }}"
+        run: |
+          go run github.com/kubeshop/botkube/hack -binaries-path "./dist"
+      - name: Publish GitHub release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create "${GITHUB_REF#refs/tags/}" \
+          --generate-notes \
+          ./dist/source_* \
+          ./dist/executor_* \
+          ./plugins-index.yaml
+```
+
+This workflow uses GoReleaser that was described in the [**Build plugin binaries**](custom-executor.md#build-plugin-binaries) section.
+
+## GitHub pages
+
+GitHub allows you to serve static pages via GitHub Pages. When you generate the index file, specify the `-url-base-path` flag as `https://{user}.github.io/{repository}`, for example, `https://kubeshop.github.io/botkube-plugins`.
+
+**Initial setup**
+
+1. Navigate to the Git repository with your plugins.
+
+2. Create the `gh-pages` branch:
```

Comment:
Since we already encouraged them to use Github actions in Go release section (previous bullet), we can also encourage them to use [gh pages github action](https://github.com/peaceiris/actions-gh-pages). Just an idea.